### PR TITLE
Add ctrl-M

### PR DESCRIPTION
**Pull request by nuess0r**
## Description
Add ctrl-M replacement controller for the IBM Model M 101/102

Detailed description of the ctrl-M controller:
https://brain4free.org/wiki/doku.php/elektronik:ctrl_m

## QMK Pull Request

Link to merged QMK Pull Request:

https://github.com/qmk/qmk_firmware/pull/24375

## VIA Keymap Pull Request

VIA keymap PR:

https://github.com/the-via/qmk_userspace_via/pull/40

## Checklist

- [x] The VIA support for this keyboard is **MERGED** in QMK master already **(MANDATORY)**
- [ ] VIA keymap is **MERGED** in VIA userspace master already **(MANDATORY)**
- [x] The VIA definition follows the guide here: https://caniusevia.com/docs/layouts
- [x] I have a V3 JSON version for this keyboard definition.**(MANDATORY)**
- [x] I have formatted the JSON file to have consistent formatting with the rest of the repository.
- [x] I have tested this keyboard definition using VIA's "Design" tab.
- [x] I have tested this keyboard definition with firmware on a device.
- [x] I have assigned alpha keys and modifier keys with the correct colors.
- [x] The Vendor ID is not `0xFEED`

### CODE REVIEW

File: src/ibm/model_m/ctrl_m/ctrl_m.json [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+          "x": 0.25,
+          "c": "#cccccc",
+          "w": 2
+        },
+        "0,12",
+        "0,13"
+      ]
+    ]
+  }
+}
```
Comment (Cipulot) [inline src/ibm/model_m/ctrl_m/ctrl_m.json:265]:
```suggestion
{
  "name": "ctrl-M",
  "vendorId": "0x1D50",
  "productId": "0x6180",
  "matrix": {
    "rows": 8,
    "cols": 16
  },
  "layouts": {
    "labels": [
      "ISO Enter"
    ],
    "keymap": [
      [
        {
          "x": 2.5,
          "c": "#777777"
        },
        "0,2",
        {
          "x": 1
        },
        "2,3",
        "2,4",
        "1,4",
        "0,4",
        {
          "x": 0.5
        },
        "0,6",
        "0,8",
        "1,9",
        "2,9",
        {
          "x": 0.5
        },
        "2,6",
        "3,6",
        "3,11",
        "3,12",
        {
          "x": 0.25
        },
        "3,15",
        "4,15",
        "6,14"
      ],
      [
        {
          "y": 0.5,
          "x": 2.5,
          "c": "#cccccc"
        },
        "2,2",
        "3,2",
        "3,3",
        "3,4",
        "3,5",
        "2,5",
        "2,7",
        "3,7",
        "3,8",
        "3,9",
        "3,10",
        "2,10",
        "2,8",
        {
          "c": "#777777",
          "w": 2
        },
        "1,6",
        {
          "x": 0.25
        },
        "2,12",
        "2,14",
        "2,13",
        {
          "x": 0.25
        },
        "6,11",
        {
          "c": "#cccccc"
        },
        "6,12",
        "6,13",
        "7,13"
      ],
      [
        {
          "x": 2.5,
          "c": "#777777",
          "w": 1.5
        },
        "1,2",
        {
          "c": "#cccccc"
        },
        "4,2",
        "4,3",
        "4,4",
        "4,5",
        "1,5",
        "1,7",
        "4,7",
        "4,8",
        "4,9",
        "4,10",
        "1,10",
        "1,8",
        {
          "w": 1.5
        },
        "5,6\n\n\n0,0",
        {
          "x": 0.25,
          "c": "#777777"
        },
        "2,11",
        "3,14",
        "3,13",
        {
          "x": 0.25,
          "c": "#cccccc"
        },
        "4,11",
        "4,12",
        "4,13",
        {
          "h": 2
        },
        "4,14",
        {
          "x": 1.25,
          "c": "#777777",
          "w": 1.25,
          "h": 2,
          "w2": 1.5,
          "h2": 1,
          "x2": -0.25
        },
        "6,6\n\n\n0,1"
      ],
      [
        {
          "x": 2.5,
          "w": 1.75
        },
        "1,3",
        {
          "c": "#cccccc"
        },
        "5,2",
        "5,3",
        "5,4",
        "5,5",
        "0,5",
        "0,7",
        "5,7",
        "5,8",
        "5,9",
        "5,10",
        "0,10",
        {
          "c": "#777777",
          "w": 2.25
        },
        "6,6\n\n\n0,0",
        {
          "x": 3.5,
          "c": "#cccccc"
        },
        "1,11",
        "1,12",
        "1,13",
        {
          "x": 1.25
        },
        "6,10\n\n\n0,1"
      ],
      [
        {
          "c": "#aaaaaa",
          "w": 1.25
        },
        "1,1\n\n\n0,1",
        {
          "c": "#cccccc"
        },
        "0,3\n\n\n0,1",
        {
          "x": 0.25,
          "c": "#aaaaaa",
          "w": 2.25
        },
        "1,1\n\n\n0,0",
        {
          "c": "#cccccc"
        },
        "6,2",
        "6,3",
        "6,4",
        "6,5",
        "7,5",
        "7,7",
        "6,7",
        "6,8",
        "6,9",
        "7,10",
        {
          "c": "#aaaaaa",
          "w": 2.75
        },
        "6,1",
        {
          "x": 1.25,
          "c": "#777777"
        },
        "0,14",
        {
          "x": 1.25,
          "c": "#cccccc"
        },
        "5,11",
        "5,12",
        "5,13",
        {
          "c": "#777777",
          "h": 2
        },
        "5,14"
      ],
      [
        {
          "x": 2.5,
          "c": "#aaaaaa",
          "w": 1.25
        },
        "2,0",
        {
          "x": 1.25,
          "w": 1.25
        },
        "0,15",
        {
          "c": "#cccccc",
          "w": 6.25
        },
        "7,6",
        {
          "c": "#aaaaaa",
          "w": 1.25
        },
        "7,15",
        {
          "x": 2.5,
          "w": 1.25
        },
        "6,0",
        {
          "x": 0.25,
          "c": "#777777"
        },
        "7,14",
        "7,11",
        "7,12",
        {
          "x": 0.25,
          "c": "#cccccc",
          "w": 2
        },
        "0,12",
        "0,13"
      ]
    ]
  }
}

```

**Review by Cipulot — APPROVED** [review]

**Comment by Cipulot** [comment]
@nuess0r pull the changes I applied to format the file and please move the file form the `src` directory to `v3`

**Comment by nuess0r** [comment]
@Cipulot thanks for the review. I moved it now to the correct folder. Either I missed the correct folder in the via documentation or it was too small that I recognized it. Sorry for that :-)

**Comment by Cipulot** [comment]
@nuess0r thx, dw about it, looks all good now, merging.

**Pull request merged**